casbin/enforcer_cached.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2020 The casbin Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CASBIN_CPP_ENFORCER_CACHED
+#define CASBIN_CPP_ENFORCER_CACHED
+
+#include <mutex>
+
+#include "./enforcer.h"
+
+class CachedEnforcer {
```

Comment:
Inherit Enforcer class, so that you can call API functions directly like this, `e.RemovePolicy()` instead of using `e.e->RemovePolicy()`

Reply:
OK,I will overwrite all the API of Enforcer.